### ISSUE TITLE
Proposal: @XXXArgsConstructor - handle automatically OR have an attribute `throws` for specifying Exception classes to declare as thrown by the generated constructor.

### ISSUE THREAD

**Issue by wheredevel** [projectlombok/lombok#1921]
```
class A {
  A() throws Exception {}
}

//Vanilla
class B {
  B() throws Exception {
    super(); // called implicitly, and it throws..! 
  }

}

//Lombok
@NoArgsConstructor(throws={Exception.class})
class B {
}
```

It might be handled automatically WITHOUT specifying such an attribute, as it's obvious: if B would not declare throws - it wouldn't compile as Vanilla case shows.  

Currently, if you put ```@NoArgsConstructor``` on B in it would give an error: "Unhandled exception of type Exception" - and you go and boiler-plate all the siblings.

**Comment by rzwitserloot** [comment]
Well, 'throws' is a keyword so it can't be (throws=...).

Sooo.. hurls=? pitches, lobs, tosses, propels, chucks?

Maybe throwables=? or exceptions=?

I don't like any of those.

If no good ideas forthcoming by 2018-12-01, close issue.

**Comment by wheredevel** [comment]
Take `throwing={Exception.class}`

**Comment by rspilker** [comment]
I know I marked this issue as `accepted`, but now I've changed my mind. 

1. Currently, all constructors we generate cannot pass parameters to a super constructor. So only the no-args super constructor will ever be invoked.

2. The most important reason to add `throws` clauses would be because the super constructor would throw a checked exception.

3. We think it is very unlikely that a no-args super constructor would throw a checked exception

4. We never generate checked exceptions in a generated constructor

So the feature does not seem to carry it's weight. Too much maintenance costs for too little benefits.
